docs/source/format/CanonicalExtensions.rst

```diff
@@ -283,6 +283,132 @@ UUID
+  return those bytes to the application, which may be able to parse the data
+  itself.  Unknown differentiates the column from an actual binary column.
+
+  An example of the extension metadata would be::
+
+    {"type_name": "geometry", "vendor_name": "PostGIS"}
+
+* The ADBC PostgreSQL driver may also get bytes for a field whose type it can
+  only partially recognize.  For example, PostgreSQL supports `composite types
+  <https://www.postgresql.org/docs/current/rowtypes.html>`_ that ascribe new
+  semantics to existing types, somewhat like Arrow extension types.
+
+  The driver would be able to parse the underlying type in this case.
+  However, the driver may still want to use the Unknown type.  Consider the
+  example in the PostgreSQL documentation above of a ``complex`` type.  Just
+  mapping the type to a plain Arrow ``struct`` type would lose the semantics
+  of that custom type.  In this case, the driver can use Unknown[Struct].  The
+  driver would never actually be able to directly support the type in this
+  example, since these types are defined by database administrators, not by
+  the developers.
+
+  An example of the extension metadata would be::
+
+    {"type_name": "database_name.schema_name.complex", "vendor_name": "PostgreSQL"}
+
+* The JDBC adapter in the Arrow Java libraries converts JDBC result sets into
+  Arrow arrays, and also to get Arrow schemas from result sets.  JDBC,
+  however, allows drivers to return `arbitrary Java objects
+  <https://docs.oracle.com/javase/8/docs/api/java/sql/Types.html#OTHER>`_.
+
+  Without the extension type, the JDBC adapter would simply error, making the
+  adapter a minefield where results are all-or-nothing, even if an application
+  just wants a schema.  Instead, the driver could use Unknown[Null] as a
+  placeholder during schema conversion, only erroring if the application tries
+  to fetch the actual data.  That way, clients could at least introspect
+  tables and queries to decide whether it can proceed to query the data, or
+  only query certain columns.
+
+  An example of the extension metadata would be::
+
+    {"type_name": "OTHER", "vendor_name": "JDBC driver name"}
+
+Of course, the intermediate system *could* implement custom extension types in
+these cases.  But there is no way that every type can be known in advance, as
+discussed specifically for each example.  In such cases, the Unknown type
+allows the system to explicitly note that it does not support some type or
+field, without silently losing data or sending irrelevant errors.
+
+Another option would be to pretend to support the type and make up new
+extension types on the fly.  But this misleads downstream systems who cannot
+tell if the type is truly supported or not by the intermediate Arrow
+application (the Flight SQL service, or the JDBC adapter, etc.), which
+particularly matters for the Flight SQL and JDBC examples.
+
+Applications **should not** make conventions around vendor_name and type_name.
```

Comment:
IMO, it might pay to move this and the next paragraph into a section above, and refer to examples of what good uses look like.

Reply:
Updated again, though I still think starting the discussion with the mindset of MIME types and JSON has distracted people and inflated the required prose a lot since this type is now being approached from an irrelevant angle.